gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
 
   private void parseConfig(Map<String, ?> config) {
     if (config != null) {
-      Boolean value = JsonUtil.getBoolean(config, "enable_cloud_logging");
-      if (value != null) {
-        enableCloudLogging = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_monitoring");
-      if (value != null) {
-        enableCloudMonitoring = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_trace");
-      if (value != null) {
-        enableCloudTracing = value;
-      }
-      destinationProjectId = JsonUtil.getString(config, "destination_project_id");
-      flushMessageCount = JsonUtil.getNumberAsLong(config, "flush_message_count");
-      List<?> rawList = JsonUtil.getList(config, "log_filters");
-      if (rawList != null) {
-        List<Map<String, ?>> jsonLogFilters = JsonUtil.checkObjectList(rawList);
-        ImmutableList.Builder<LogFilter> logFiltersBuilder = new ImmutableList.Builder<>();
-        for (Map<String, ?> jsonLogFilter : jsonLogFilters) {
-          logFiltersBuilder.add(parseJsonLogFilter(jsonLogFilter));
-        }
-        this.logFilters = logFiltersBuilder.build();
+      projectId = JsonUtil.getString(config, "project_id");
+      parseLoggingObject(config, CONFIG_LOGGING_VAR_NAME);
+      parseMonitoringObject(config, CONFIG_MONITORING_VAR_NAME);
+      parseTracingObject(config, CONFIG_TRACING_VAR_NAME);
+      parseCustomTags(config, CONFIG_LABELS_VAR_NAME);
+    }
+  }
+
+  private void parseLoggingObject(Map<String, ?> config,  String jsonObjectName) {
+    Map<String, ?> rawCloudLogging = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudLogging != null) {
+      enableCloudLogging = true;
+      List<?> rawClientRpcEvents = JsonUtil.getList(rawCloudLogging, "client_rpc_events");
+      clientLogFilters =
+          rawClientRpcEvents != null ? parseRpcEvents(rawClientRpcEvents) : clientLogFilters;
+      List<?> rawServerRpcEvents = JsonUtil.getList(rawCloudLogging, "server_rpc_events");
+      serverLogFilters =
+          rawServerRpcEvents != null ? parseRpcEvents(rawServerRpcEvents) : serverLogFilters;
+    }
+  }
+
+  private void parseMonitoringObject(Map<String, ?> config, String jsonObjectName) {
+    Map<String, ?> rawCloudMonitoring = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudMonitoring != null) {
+      if (rawCloudMonitoring.isEmpty()) {
```

Comment:
What is with this check?

Reply:
Added this check for validation, if there are any nested objects and fields we might not want to enable monitoring. But as per new design, having an empty object enabled monitoring.  I will remove this check.